Bacterial Genome Assembly with Unicycler Training

Delphine-L opened the pull request:
Hello, 
I am working on a Small genome Genome assembly training. This pull request is just here for information purpose, the material is still a work in progress. 
I would be grateful for any feedback on the material.

bgruening commented:
@Delphine-L this is a very nice start! Thanks a lot for working on it!

willdurand commented on topics/assembly/tutorials/unicycler-assembly/tutorial.md:
```diff
@@ -108,64 +113,69 @@ Prokka predicts protein-coding regions using a two step process. It first identi
 {: .agenda}
 
-### <a name="get-the-data">Get the data
+## <a name="get-the-data">Load data and assess quality
```
I don't think the anchor is useful anymore. There should be an anchor on each title.

willdurand commented on topics/assembly/metadata.yaml:
```diff
@@ -63,11 +57,39 @@ material:
+    key_points:
 
 maintainers:
-  - slugger70
```
We put usernames only here now and there is a CONTRIBUTORS file with all the information.

nekrut replied:
so, `contributors` goes away from this file?

shiltemann replied:
it stays in, but only the username of the contributor/maintainer should be given (like this removed line), not the full name/email etc like it used to be. This name is then matched to the information in the `CONTRIBUTORS.yaml` file in the root of the repo, where information such as full name and email address can be associated with the user

nekrut commented:
@Delphine-L give it a read!

bgruening commented on topics/assembly/metadata.yaml:
```diff
@@ -63,12 +57,32 @@ material:
+    time_estimation: "4h"
+    key_points:
     contributors:
-      - slugger70
```
I guess we want to keep Simon here.

bgruening commented:
@Delphine-L is this ready to go or still WIP?

Delphine-L commented:
@bgruening Im running the docker image to verify the tour one last time and it will be ready

Delphine-L commented:
@bgruening It's ready to go !

bgruening commented:
Give me a few minutes to check again - but I'm so motivated to get this in I will hurry up!

Thanks a lot!

shiltemann commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1143 @@
+
+Previously we have learned about applications on PacBio technology:
+
+{{< speakerdeck 5850fe7ede3b45df816c3eaa502ea3d1 >}}
```
what is the intension of this line?

shiltemann commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1143 @@
+---
+layout: introduction_slides
+topic_name: unicycler_assembly
```
topic should be `assembly`

shiltemann commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1143 @@
+---
+layout: introduction_slides
+topic_name: unicycler_assembly
+tutorial_name: am_genome_assembly
```
tutorial name should match the name in the metadata file, so `unicycler-assembly`

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+
+**Sources of this Training:**
+
+Assembly basics Topic 12 : Challenges of genome (and transcriptome) assembly <br>
```
Is this here needed, or is a link missing?

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
````diff
@@ -0,0 +1,1142 @@
+Read X CTCTAGGCC
+Read Y TAGGCCCTC
+```
+what is the overlap between these two reads? For now we will define overlap of <code>length - l</code> suffix of Read X matches <code>length - l</code> prefix of Read Y, where <code>l</code> is given. To find these overlap we look in Read Y for instances <code>length - l</code> suffix of Read X.
````
s/what/What/

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+
+---
+
+As a result we represent two two reads are connected nodes:
```
s/two two/two/

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+BAA AAB BBA ABA ABB BBB AAA BAB
+</samp>
+
+one way of getting a string that would contain all of these as substrings will simply be concatenating them:
```
One

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+</samp>
+
+
+this, however, is not the *shortest* superstring that contains all strings from $S$. Instead the SCS is (just trust us here):
```
This

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+
+As we've seen above the shortest common superstring (SCS) is:
+
+1. **Difficult to obtain** as Greedy SCS algorithm does not guarantee funding it. So the answer we get may be longer that the real genome we are trying to assemble.
```
s/funding/finding/

bgruening replied:
s/that/than/

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+
+---
+
+Let's apply Euler's Theorem to a classical problem: The bridges of Köninsberg problem. Here the question is: *Can you walk through all of Köninsberg traversing every bridge exactly one time?* In other words: *Is there a Eulerian path through the city of Köninsberg?*
```
Its "Königsberg"

bgruening commented on topics/assembly/tutorials/unicycler-assembly/slides.html:
```diff
@@ -0,0 +1,1142 @@
+
+**Ploidy**
+
+As we discussed earlier humans are, for example, diploid and there are multiple differences between maternal and paternal genomes. This creates "bubbles" on assembly graphs:
```
humans are, for example, diploid ? :)

bgruening commented on topics/assembly/tutorials/unicycler-assembly/tours/tour.yaml:
```diff
@@ -0,0 +1,218 @@
+      position: "center"
+      
+    - title : "Search a Tool"
+      content: "Search for a tool to evaluate read quality : Type FastQC in the search field."
```
quality :  -> quality:

bgruening commented on topics/assembly/tutorials/unicycler-assembly/tours/tour.yaml:
```diff
@@ -0,0 +1,218 @@
+              Hit the close button when you see that the files are uploaded into your history."
+      position: "bottom"
+
+    - title : "Start the Analysis !"
```
maybe this is style, but I prefer !, :, ? directly after a word.

bgruening commented on topics/assembly/tutorials/unicycler-assembly/tours/tour.yaml:
```diff
@@ -0,0 +1,218 @@
+      element : "#execute"
+
+    - title: "Read Quality "
+      content: "Once FastQC is done running (green),you can visualize the Quality control result of each file by clicking on the eye icon of the Fastqc webpage output. For more information on the quality metrics of FastQC, see <a href=\"https://biof-edu.colorado.edu/videos/dowell-short-read-class/day-4/fastqc-manual\"> FastQC Manual."
```
,<space>

bgruening replied:
quality control

bgruening commented:
@Delphine-L a few comments included, but really just small things. Very well done!

Delphine-L commented:
Thank you @bgruening ! I made the corrections.

bebatut commented on topics/assembly/metadata.yaml:
```diff
@@ -65,10 +65,33 @@ material:
+      - "Evaluate the Quality of the Assembly with Quast"
+      - "Annotate the assembly with Prokka"
+    time_estimation: "4h"
+    key_points:
```
Can you add some key-points or take-home messages? They help trainees (and also instructors :wink: ) at the end of the tutorial to sum up what learned during the tutorial

bebatut commented on topics/assembly/tutorials/unicycler-assembly/tutorial.md:
```diff
@@ -0,0 +1,268 @@
+
+To load data into your Galaxy instance log in into Galaxy and create new history (if you are new to Galaxy see [Galaxy 101 tutorial](/topics/introduction/tutorials/galaxy-intro-101/tutorial.html) first). 
+
+Click **Get data** icon as shown below (see [these slides](/topics/introduction/tutorials/galaxy-intro-get-data/slides.html) for an introduction on how to load data into Galaxy):
```
To make the hands-on sections more visible during workshops and self-training, we decided to put them into boxes like:
![screen shot 2017-10-05 at 19 04 03](https://user-images.githubusercontent.com/1842467/31240915-798b0f9c-aa02-11e7-9bea-00cdd3d92080.png)

Can you add the hands-on sections of your tutorial into such boxes? Something that looks like:

```
> ### {% icon hands_on %} Hands-on: Align sequences
>
> 1. Import the `silva.v4.fasta` file in your history
> 2. **Align.seqs** {% icon tool %} with the following parameters
>   - "fasta" to the `good.fasta` output from `Screen.seqs`
>   - "reference" to the `silva.v4.fasta` reference file
>   - "flip" to `Yes`
>
>    This step may take a few minutes, please be patient.
>
> 3. **Summary.seqs** {% icon tool %} with the following parameters
>   - "fasta" parameter to the aligned output from `Align.seqs`
>   - "count" parameter to count_table output from `Screen.seqs`
>
{: .hands_on}
```

bebatut commented on topics/assembly/tutorials/unicycler-assembly/tutorial.md:
```diff
@@ -0,0 +1,268 @@
+> Here we assume that you know a thing or two about assembly process. If you don't: look at the [slides](./slides) accompanying this tutorial as well as other tutorials is this section.
+{: .info-box}
+
+![](https://github.com/rrwick/Unicycler/raw/master/misc/logo.png)
```
Can you add an alternative text between the `[]`? The test is complaining about that

shiltemann commented on assets/css/main.scss:
```diff
@@ -11,6 +11,12 @@ $overview-color: $bg-color;
+$warning-color: #FAA632;
+$info-color: #4aafcd;
+$danger-color: #da4f4a;
+$success-color: #5ab75c;
```
I personally really love these more colourful boxes! 

However, in the interest of keeping the tutorials looking uniform, I think we should try to minimize the number of different types of boxes we have, and instead re-use the existing box types where possible, and then maybe just rethink the colour palette/icon set for these at some point? For instance, perhaps we can re-use the `comment` box instead of the new `info` (and maybe `warning`) box to give the extra/background information to users? What do you think? 

Also, it looks like you don't use the `success` and `danger` boxes in your tutorial, could we remove those from the template for now?

shiltemann commented:
@Delphine-L I would say that if we list all the parameter settings clearly in text form (like the example code), that we maybe do not even need the screenshots at all (and this is also better for accessibility, and easier to update in the future). what do you think? (and otherwise I would say leave them inside the boxes because they are part of the hands-on segment)

Delphine-L commented:
@shiltemann  I removed the unused boxes and I replace the info box in the tutorial by a comment box. I changed the color of the comment box then. I also removed the pictures that were showing the tool parameters , replacing it by the description, but I let those that was showing less straight forward manipulations, I think it's more readable that way. What do you think?

shiltemann commented:
@Delphine-L sounds good to me, thanks! we can think about the color scheme of the other box types too in some other PR then, if you feel more distinct colors are useful (I agree). Sorry to have been nitpicky, this really is an awesome tutorial, thanks a lot for making it!

Delphine-L commented:
Thanks @bebatut ! I removed the hr tags and lines around images, it's more consistant that way. Do you think it's ready to be merged?

nekrut commented:
@bebatut or @shiltemann : pls, merge as these errors don't make any sense

bebatut commented on topics/assembly/tutorials/unicycler-assembly/tutorial.md:
```diff
@@ -0,0 +1,272 @@
+> 
+> 4. Let's add Prokka annotations to the browser image. For this simply expand Prokka's GFF3 dataset and click on the *local* link:
+> 
+>      ![Expanded GFF3 dataset representing Prokka annotations](../../images/prokka_item.png "Expanded GFF dataset generated with Prokka. Click on the <em>local</em> link (highlighted with orange outline) to display this dataset within IGV.")ßßß
```
Are the `ßßß` normal or did you add them by mistake?

Delphine-L replied:
It was a mistake, I removed it. Thanks for seing it !

bebatut commented on topics/assembly/tutorials/unicycler-assembly/tutorial.md:
```diff
@@ -0,0 +1,272 @@
+
+### Assembly
+
+> ### <i class="fa fa-lightbulb-o" aria-hidden="true"></i> Knowing your assembly
```
Can you replace the `<i class="fa fa-lightbulb-o" aria-hidden="true"></i>` by `{% icon comment %}`? The list of icons for the boxes can be found in `_config.yml` 
You can also add the icon for the warning box before (I will do it otherwise after merging the PR) :smile:

bebatut replied:
I can do it also afterwards (changing the icons) 😄

Delphine-L replied:
I replaced the comment icon and added the warning icon to _config.xml (the warning sign).

nekrut commented:
@bebatut @Delphine-L @shiltemann errors are these:
```
- ./_site/topics/assembly/index.html
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/slides.html failed: 404 No error
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html failed: 404 No error
- ./_site/topics/assembly/tutorials/unicycler-assembly/tutorial.html
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/slides failed: 404 No error
  *  External link https://github.com/galaxyproject/training-material/tree/master/topics/assembly/tutorials/unicycler-assembly/tutorial.md failed: 404 No error
htmlproofer 3.7.4 | Error:  HTML-Proofer found 4 failures!
make: *** [check] Error 1
```
Ok to merge?

bebatut commented:
I was afraid to do it 😄

bebatut commented:
https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html 🎉

shiltemann commented:
\o/ :tada: thanks a lot @Delphine-L @nekrut !
